### PR TITLE
merge `no-unused-import` and `no-unused-vars`

### LINKED ISSUE #23: Merge no-unused-import and no-unused-var
Currently no-unused-var doesn't track imports and no-unused-import is not aware of name shadowing. Given that also I can't imagine a scenario where someone would want to have one rule in use but not the other, and also considering the discussion in #3 , I say we merge the two rules for v4.

### LINKED ISSUE #24: Recognize shadowing in no-unused-import rule
Failing tests for this issue in PR #22

Currently, the no-unused-import rule considers usages of Identifiers with a name defined in an
ImportDirective to be usages of said name. But there's an edge case where an imported is not used,
but there's a variable shadowing its name. Currently, this causes a false negative.

Ideally, the unused import would be reported, and we'd have a rule forbidding name shadowing as
well.

I consider it not too bad of a bug, since solc usually warns about name shadowing and it's not a
frequent scenario anyway, so my intention is to leave it be for the time being. 

The most lazy way we can go about this is to fix this issue when [we merge no-unused-var and
no-unused-import](https://github.com/solhint-community/solhint-community/issues/23) in v4

### PR DESCRIPTION
**Pull request by juanpcapurro**
closes #24 
closes #23 

see commit history for details

### CODE REVIEW

File: lib/rules/best-practises/no-unused-vars.js [hunk 1]
```diff
@@ -71,14 +71,15 @@ class NoUnusedVarsChecker extends BaseChecker {
   }
 
   'FunctionDefinition:exit'(node) {
+    // we skip adding a var usage scope for function _declarations_ and
+    // functions with an empty / without a code block
```
Comment (juanigaray) [inline lib/rules/best-practises/no-unused-vars.js]: hard to parse as sth else than "functions with an empty slash that do not have a code block". Would rephrase

Comment (juanigaray) [reply lib/rules/best-practises/no-unused-vars.js]: resolve at will btw
